File: gescom_angular/src/app/produits/produit.ts
import {Categorie} from "./categorie";

export class Produit {
  id: number = 0;
  nom: string | undefined;
  designation: string | undefined;
  prix: number | undefined;
  stock: number | undefined;
  imageUrl: string | undefined;
  categorie: Categorie | undefined;


}

File: java/gestionStock_springBoot/src/main/java/com/edugroupe/gestionstock_springboot/entity/Produit.java
package com.edugroupe.gestionstock_springboot.entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Objects;

@Entity
@Table(name = "produit")
@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@ToString
@EqualsAndHashCode
public class Produit implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String nom;
    private String designation;
    private double prix;
    private int stock;
    @Column(name = "image_url")
    private String imageUrl;
    @ManyToOne
    @JoinColumn(name = "categorie_id")

    private Categorie categorie;





}
